lib/configParser.ts

```diff
@@ -0,0 +1,230 @@
+import * as path from 'path';
+import * as glob from 'glob';
+import * as logger from './logger'
+
+// Coffee is required here to enable config files written in coffee-script.
+try {
+  require('coffee-script').register();
+} catch (e) {
+  // Intentionally blank - ignore if coffee-script is not available.
+}
+
+// LiveScript is required here to enable config files written in LiveScript.
+try {
+  require('LiveScript');
+} catch (e) {
+  // Intentionally blank - ignore if LiveScript is not available.
+}
+
+export class ConfigParser {
+  public constructor() {}
+
+  // Default configuration.
+  private config_ = {
+    specs: [],
+    multiCapabilities: [],
+    rootElement: 'body',
+    allScriptsTimeout: 11000,
+    getPageTimeout: 10000,
+    params: {},
+    framework: 'jasmine',
+    jasmineNodeOpts: {
+      showColors: true,
+      // stackFilter: helper.filterStackTrace,
+      defaultTimeoutInterval: (30 * 1000)
+    },
+    seleniumArgs: [],
+    seleniumSessionId: null,
+    mochaOpts: {
+      ui: 'bdd',
+      reporter: 'list'
+    },
+    chromeDriver: null,
+    configDir: './',
+    plugins: [],
+    skipSourceMapSupport: false
+  };
+
+  /**
+   * Resolve a list of file patterns into a list of individual file paths.
+   *
+   * @param {Array.<string> | string} patterns
+   * @param {=boolean} opt_omitWarnings Whether to omit did not match warnings
+   * @param {=string} opt_relativeTo Path to resolve patterns against
+   *
+   * @return {Array} The resolved file paths.
+   */
+  public static resolveFilePatterns(patterns, opt_omitWarnings, opt_relativeTo) {
```

Comment:
if you add --noImplicitAny to your compiler options, you'll get typechecker errors to add types here. But you could do it later since this is an incremental migration